src/Microsoft.TestPlatform.TestHostProvider/Hosting/DotnetTestHostManager.cs

```diff
@@ -408,7 +408,7 @@ public Task<bool> LaunchTestHostAsync(TestProcessStartInfo testHostStartInfo, Ca
 
             // We silently force x64 only if the target architecture is the default one and is not specified by user
             // through --arch or runsettings or -- RunConfiguration.TargetPlatform=arch
-            bool forceToX64 = SilentlyForceToX64() && _runsettingHelper.IsDefaultTargetArchitecture;
+            bool forceToX64 = SilentlyForceToX64(sourcePath) && _runsettingHelper.IsDefaultTargetArchitecture;
```

Comment:
Is this in this order on purpose? SilentlyForceToX64 should be evaluated even if IsDefaultTargetArchitecture is true?

Reply:
> Is this in this order on purpose? SilentlyForceToX64 should be evaluated even if IsDefaultTargetArchitecture is true?

do you meant `false`